ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala

```diff
@@ -117,43 +121,34 @@ final class CommandService(session: LedgerSession) extends LedgerTestSuite(sessi
 
   private val resendingSubmitAndWait = LedgerTest(
     "CSduplicateSubmitAndWait",
-    "SubmitAndWait should be idempotent when reusing the same command identifier") {
-    implicit context =>
-      val duplicateCommandId = "CSduplicateSubmitAndWait"
-      for {
-        alice <- allocateParty()
-        _ <- submitAndWait(
-          alice,
-          Dummy(alice).create.command,
-          _.commands.commandId := duplicateCommandId)
-        _ <- submitAndWait(
-          alice,
-          Dummy(alice).create.command,
-          _.commands.commandId := duplicateCommandId)
-        transactions <- flatTransactions(alice)
-      } yield {
-        assert(
-          transactions.size == 1,
-          s"Expected only 1 transaction, but received ${transactions.size}")
+    "SubmitAndWait should be idempotent when reusing the same command identifier") { ledger =>
+    val duplicateCommandId = "CSduplicateSubmitAndWait"
+    for {
+      alice <- ledger.allocateParty()
+      request <- ledger.submitAndWaitRequest(alice, Dummy(alice).create.command)
+      duplicate = request.update(_.commands.commandId := duplicateCommandId)
```

Comment:
Is it not enough to simply send the same request twice? I don't think we need a special command id for that.

Reply:
Ah, sure.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2573/commits/3b3c89b38b09cd6081c1dad5f58e13b03ff24fb9